paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/engine/PaddlePredictor.java

```diff
@@ -10,25 +10,26 @@
  * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
  * and limitations under the License.
  */
+package ai.djl.paddlepaddle.engine;
 
-package ai.djl.paddlepaddle.jna;
+import ai.djl.paddlepaddle.jni.JniUtils;
+import ai.djl.util.NativeResource;
 
-import com.sun.jna.IntegerType;
-import com.sun.jna.Native;
+/** PaddlePaddle C++ Predictor. */
+public class PaddlePredictor extends NativeResource<Long> {
 
-/** The Paddle Native size handler. */
-@SuppressWarnings("missingjavadocmethod")
-public class NativeSize extends IntegerType {
-
-    private static final long serialVersionUID = 1L;
-
-    public static final int SIZE = Native.SIZE_T_SIZE;
+    PaddlePredictor(long handle) {
```

Comment:
can it be a simple JniUtils function call rather than a class?

Reply:
We have two options. Make it as a long and do all raw use including its life cycle. Or we wrap it up as an object.

Reply:
the line of code seems to be almost the same. let's just keep it as it is